ckanext/canada/extract.py

```diff
@@ -0,0 +1,86 @@
+from yaml import load
+from yaml.loader import SafeLoader
+from six import string_types
+
+
+class SafeLineLoader(SafeLoader):
+    def __init__(self, stream):
+        super(SafeLineLoader, self).__init__(stream)
+        self.line_numbers = {}
+
+    def construct_scalar(self, node):
+        if node.value in self.line_numbers:
+            self.line_numbers[node.value].append(node.start_mark.line + 1)
+        else:
+            self.line_numbers[node.value] = [node.start_mark.line + 1]
+        return super(SafeLineLoader, self).construct_scalar(node)
+
+    def get_single_data(self):
+        data = super(SafeLineLoader, self).get_single_data()
+        return self.line_numbers, data
```

Comment:
Great, this is much nicer than searching across the yaml file to try to locate strings.